File: UnitTestEx/UnitTestEx/UnitTestProject/FileTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using NUnit.Framework;
using System;
using UnitTestEx;
using Assert = NUnit.Framework.Assert;

namespace UnitTestProject
{
    [TestClass]
    public class FileTest
    {

        public const string SIZE_EXCEPTION = "Wrong size";
        public const string NAME_EXCEPTION = "Wrong name";
        public const string SPACE_STRING = " ";
       // public const string FILE_PATH_STRING = "@D:\\JDK-intellij-downloader-info.txt";
        public const string CONTENT_STRING = "Some text";
        public const string NAME_STRING = "AA";
        public double lenght;

        /* ПРОВАЙДЕР */
        //static object[] FilesData =
        //{
        //    new object[] {new File(FILE_PATH_STRING, CONTENT_STRING), FILE_PATH_STRING, CONTENT_STRING},
        //    new object[] { new File(SPACE_STRING, SPACE_STRING), SPACE_STRING, SPACE_STRING}
        //};

        /* Тестируем получение размера */
        [TestMethod]
        public void GetSizeTest()
        {
            //arrange
            File file = new File(NAME_STRING, CONTENT_STRING);
            //act
            double size = file.GetSize();
            lenght = CONTENT_STRING.Length / 2;
            //assert
            Assert.AreEqual(lenght,size);
            
        }

        /* Тестируем получение имени */
        [TestMethod]
        public void GetFilenameTest()
        {
            //arrange
            File file=new File(NAME_STRING, CONTENT_STRING);
            //act and assert
            Assert.AreEqual(NAME_STRING,file.GetFilename());
        }

    }
}

File: Task1/src/test/java/FileTest.java
import files.File;
import org.testng.Assert;
import org.testng.annotations.DataProvider;
import org.testng.annotations.Test;

public class FileTest {
    public final String SIZE_EXCEPTION = "Wrong size";
    public final String NAME_EXCEPTION = "Wrong name";
    public final String SPACE_STRING = " ";
    public final String FILE_PATH_STRING = "@D:\\JDK-intellij-downloader-info.txt";
    public final String CONTENT_STRING = "Some text";

    public double lenght;

    /* ПРОВАЙДЕР */
    @DataProvider(name = "filesData")
    public Object[][] setFileList(){
        return new Object[][] { {new File(FILE_PATH_STRING, CONTENT_STRING), FILE_PATH_STRING, CONTENT_STRING},
                {new File(SPACE_STRING, SPACE_STRING), SPACE_STRING, SPACE_STRING} };
    }

    /* Тестируем получение размера */
    @Test (dataProvider = "filesData")
    public void testGetSize(File newFile, String name, String content) {
        lenght = content.length() / 2;
        Assert.assertEquals(newFile.getSize(), lenght, SIZE_EXCEPTION);
    }

    /* Тестируем получение имени */
    @Test (dataProvider = "filesData")
    public void testGetFilename(File newFile, String name, String content) {
        Assert.assertEquals(newFile.getFilename(), name, NAME_EXCEPTION);
    }
}